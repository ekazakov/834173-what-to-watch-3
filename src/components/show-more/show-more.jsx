import React from "react";
import {connect} from "react-redux";
import PropTypes from "prop-types";
import {getShownFilms, getHaveMoreFilms} from "../../reducer/state/selectors.js";
import {ActionCreator} from "../../reducer/state/state.js";

const ShowMore = (props) => {
  const {shownFilms, showMoreFilms, haveMoreFilms} = props;

  return haveMoreFilms ? (
    <div className="catalog__more">
      <button className="catalog__button" type="button" onClick={() => showMoreFilms(shownFilms)}>Show more</button>
    </div>
  ) : null;
};

ShowMore.propTypes = {
  shownFilms: PropTypes.number.isRequired,
  showMoreFilms: PropTypes.func.isRequired,
  haveMoreFilms: PropTypes.bool.isRequired,
};

const mapStateToProps = (state) => ({
  shownFilms: getShownFilms(state),
  haveMoreFilms: getHaveMoreFilms(state),
});

const mapDispatchToProps = (dispatch) => ({
  showMoreFilms(shownFilms) {
    dispatch(ActionCreator.showMoreFilms(shownFilms));
  },
});

export {ShowMore};
export default connect(mapStateToProps, mapDispatchToProps)(ShowMore);
