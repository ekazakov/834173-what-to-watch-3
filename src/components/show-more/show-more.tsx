import * as React from "react";
import {connect} from "react-redux";
import {getShownFilms, getHaveMoreFilms} from "../../reducer/state/selectors";
import {ActionCreator} from "../../reducer/state/state";

interface ShowMoreProps {
  shownFilms: number;
  showMoreFilms: (number) => void;
  haveMoreFilms: boolean;
}

const ShowMore: React.FunctionComponent<ShowMoreProps> = (props: ShowMoreProps) => {
  const {shownFilms, showMoreFilms, haveMoreFilms} = props;

  return haveMoreFilms ? (
    <div className="catalog__more">
      <button className="catalog__button" type="button" onClick={() => showMoreFilms(shownFilms)}>Show more</button>
    </div>
  ) : null;
};

const mapStateToProps = (state) => ({
  shownFilms: getShownFilms(state),
  haveMoreFilms: getHaveMoreFilms(state),
});

const mapDispatchToProps = (dispatch) => ({
  showMoreFilms(shownFilms) {
    dispatch(ActionCreator.showMoreFilms(shownFilms));
  },
});

export {ShowMore};
export default connect(mapStateToProps, mapDispatchToProps)(ShowMore);
