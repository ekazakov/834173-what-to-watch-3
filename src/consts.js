import PropTypes from "prop-types";

export const VIDEO_TIMER_HOVER = 1000;

export const SAME_FILMS_COUNT = 4;

export const SHOWN_FILMS_DEFAULT = 8;

export const Genres = {
  ALL: `All-genres`,
  HORROR: `horrors`,
};

export const TextCommentLength = {
  MIN: 40,
  MAX: 400,
};

export const MINUTES_IN_HOUR = 60;

export const filmProps = PropTypes.shape({
  id: PropTypes.number.isRequired,
  name: PropTypes.string.isRequired,
  posterBig: PropTypes.string.isRequired,
  poster: PropTypes.string.isRequired,
  background: PropTypes.string.isRequired,
  backgroundColor: PropTypes.string.isRequired,
  video: PropTypes.string.isRequired,
  preview: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  genre: PropTypes.string.isRequired,
  rating: PropTypes.number.isRequired,
  score: PropTypes.number.isRequired,
  director: PropTypes.string.isRequired,
  starring: PropTypes.arrayOf(PropTypes.string).isRequired,
  duration: PropTypes.number.isRequired,
  year: PropTypes.number.isRequired,
  favorite: PropTypes.bool.isRequired,
});

export const commentProps = PropTypes.shape({
  id: PropTypes.number.isRequired,
  user: PropTypes.shape({
    id: PropTypes.number.isRequired,
    name: PropTypes.string.isRequired,
  }).isRequired,
  rating: PropTypes.number.isRequired,
  comment: PropTypes.string.isRequired,
  date: PropTypes.string.isRequired,
});

export const filmsProps = PropTypes.arrayOf(filmProps).isRequired;
export const commentsProps = PropTypes.arrayOf(commentProps).isRequired;

export const AuthorizationStatus = {
  AUTH: `AUTH`,
  NO_AUTH: `NO_AUTH`,
};

export const ServerStatus = {
  AVAILABLE: `AVAILABLE`,
  NO_AVAILABLE: `NO_AVAILABLE`,
};

export const TabsName = {
  OVERVIEW: `Overview`,
  DETAILS: `Details`,
  REVIEWS: `Reviews`,
};

export const AppRoute = {
  ROOT: `/`,
  LOGIN: `/login`,
  FAVORITE: `/mylist`,
  FILM: `/films`,
  REVIEW: `/review`,
  PLAYER: `/player`,
};

export const FavoriteStatus = {
  ADD: 1,
  DELETE: 0,
};
