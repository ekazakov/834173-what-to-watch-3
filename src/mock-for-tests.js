export const films = [
  {
    id: 1,
    name: `Harry Potter and the Goblet of Fire`,
    posterBig: `https://unsplash.it/280/175/`,
    poster: `https://unsplash.it/280/175/`,
    background: `https://unsplash.it/280/175/`,
    backgroundColor: `#ffffff`,
    video: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    preview: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    description: `Самый увлекательный в мире фильм`,
    genre: `drama`,
    rating: 6.7,
    score: 290,
    director: `Tarantino`,
    starring: [`Donatello`, `Rafael`, `Leonardo`],
    duration: 90,
    year: 2067,
    favorite: false,
  },
  {
    id: 2,
    name: `EuroTrip`,
    posterBig: `https://unsplash.it/280/175/`,
    poster: `https://unsplash.it/280/175/`,
    background: `https://unsplash.it/280/175/`,
    backgroundColor: `#ffffff`,
    video: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    preview: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    description: `Самый увлекательный в мире фильм`,
    genre: `drama`,
    rating: 6.7,
    score: 290,
    director: `Tarantino`,
    starring: [`Donatello`, `Rafael`, `Leonardo`],
    duration: 90,
    year: 2067,
    favorite: false,
  },
  {
    id: 3,
    name: `The Autopsy of Jane Doe`,
    posterBig: `https://unsplash.it/280/175/`,
    poster: `https://unsplash.it/280/175/`,
    background: `https://unsplash.it/280/175/`,
    backgroundColor: `#ffffff`,
    video: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    preview: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    description: `Самый увлекательный в мире фильм`,
    genre: `drama`,
    rating: 6.7,
    score: 290,
    director: `Tarantino`,
    starring: [`Donatello`, `Rafael`, `Leonardo`],
    duration: 90,
    year: 2067,
    favorite: false
  },
  {
    id: 4,
    name: `The Notebook`,
    posterBig: `https://unsplash.it/280/175/`,
    poster: `https://unsplash.it/280/175/`,
    background: `https://unsplash.it/280/175/`,
    backgroundColor: `#ffffff`,
    video: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    preview: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    description: `Самый увлекательный в мире фильм`,
    genre: `drama`,
    rating: 6.7,
    score: 290,
    director: `Tarantino`,
    starring: [`Donatello`, `Rafael`, `Leonardo`],
    duration: 90,
    year: 2067,
    favorite: false
  },
  {
    id: 5,
    name: `Carri`,
    posterBig: `https://unsplash.it/280/175/`,
    poster: `https://unsplash.it/280/175/`,
    background: `https://unsplash.it/280/175/`,
    backgroundColor: `#ffffff`,
    video: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    preview: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    description: `Самый увлекательный в мире фильм`,
    genre: `drama`,
    rating: 6.7,
    score: 290,
    director: `Tarantino`,
    starring: [`Donatello`, `Rafael`, `Leonardo`],
    duration: 90,
    year: 2067,
    favorite: false
  },
];

export const comments = [
  {
    id: 0,
    user: {
      id: 0,
      name: `Kate`,
    },
    rating: 7.6,
    comment: `OMG`,
    date: `122`,
  },
  {
    id: 1,
    user: {
      id: 0,
      name: `David`,
    },
    rating: 7.6,
    comment: `OMG`,
    date: `122`,
  },
];
