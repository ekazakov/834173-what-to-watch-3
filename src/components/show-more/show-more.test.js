import React from "react";
import renderer from "react-test-renderer";
import {ShowMore} from "./show-more.jsx";
import {SHOWN_FILMS_DEFAULT} from "../../consts";
import {films} from "../../mock-for-tests";

const shownFilms = SHOWN_FILMS_DEFAULT;

it(`Should ShowMore render correctly`, () => {

  const tree = renderer.create(
      <ShowMore films={films} shownFilms={shownFilms} showMoreFilms={() => {}} haveMoreFilms={true}/>
  )
   .toJSON();

  expect(tree).toMatchSnapshot();
});
