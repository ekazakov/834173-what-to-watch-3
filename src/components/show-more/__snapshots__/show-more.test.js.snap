// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should ShowMore render correctly 1`] = `null`;
